src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -96,7 +97,7 @@ public function registerClasses(Definition $prototype, string $namespace, string
             throw new InvalidArgumentException(sprintf('Namespace is not a valid PSR-4 prefix: "%s".', $namespace));
         }
 
-        $classes = $this->findClasses($namespace, $resource, (array) $exclude);
+        $classes = $this->findClasses($namespace, $resource, (array) $exclude, $prototype->isAutoconfigured());
```

Comment:
The attribute is parsed only if the autodiscovery group has autoconfiguration enabled.
This extra check could be removed. Let me know how you feel about this.

Reply:
Closing as I think this is the correct behavior.